File: examples/js_toy/math.js
/**
 * Small JS fixture for Understand-First Wave 15 best-effort analyzer tests.
 */
function add(a, b) {
  return a + b;
}

function compute(x, y) {
  const total = add(x, y);
  return total;
}

function classify(n) {
  if (n < 0) {
    return "neg";
  }
  for (let i = 0; i < n; i++) {
    if (i > 10 && i < 20) {
      return "mid";
    }
  }
  try {
    return String(n);
  } catch (err) {
    return "err";
  }
}

class Calculator {
  scale(v) {
    return add(v, 0);
  }

  run(a, b) {
    return this.scale(compute(a, b));
  }
}

const double = (n) => add(n, n);

module.exports = { add, compute, classify, Calculator, double };

File: examples/rust_toy/math.rs
//! Small Rust fixture for Understand-First Wave 22.

pub fn add(a: i32, b: i32) -> i32 {
    a + b
}

pub fn compute(x: i32, y: i32) -> i32 {
    add(x, y)
}

pub fn classify(n: i32) -> &'static str {
    if n < 0 {
        return "neg";
    }
    for i in 0..n {
        if i > 10 && i < 20 {
            return "mid";
        }
    }
    match n {
        0 => "zero",
        _ => "other",
    }
}

pub struct Calc {
    pub n: i32,
}

impl Calc {
    pub fn scale(&self, v: i32) -> i32 {
        add(v, 0)
    }

    pub fn run(&self, a: i32, b: i32) -> i32 {
        self.scale(compute(a, b))
    }
}
